portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
+function startAuth() {
+  handleCommentFormVisibility();
+  initializeAuth();  
```

Comment:
Each of these functions makes a call to "/auth". You could optimize this to be only one call, and pass the promise as parameters to these 2 functions so they can add their bits of logic to it.

Reply:
I'm not sure if I understand fully. I get that you're saying it would be better to make one call to /auth but I don't get how you're saying to combine the logic of both functions

Reply:
You could, for example, do something like:

```
fetch("/auth")
  .then((response) => response.json())
  .then((loginInfo) => {
    handleCommentFormVisibility(loginInfo);
    initializeAuth(loginInfo);
  });
```

Your `handleCommentFormVisibility` and `initializeAuth` functions would take the loginInfo parameter instead of obtaining it via `fetch`.

Reply:
Ok, i see now. I'll make an issue for this so I can implement it later.